tests/test_reserveSourcesTask.py

```diff
@@ -0,0 +1,154 @@
+#
+# LSST Data Management System
+#
+# Copyright 2008-2017  AURA/LSST.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+#
+from __future__ import absolute_import, division, print_function
+
+import unittest
+import numpy as np
+
+import lsst.afw.table
+import lsst.meas.algorithms
+import lsst.utils.tests
+
+from lsst.pipe.base import Struct
+
+
+class ReserveSourcesTaskTest(lsst.utils.tests.TestCase):
+    """TestCase for the ReserveSourcesTask"""
+    def setUp(self):
+        self.num = 100  # Number of sources
```

Comment:
You should also set `self.longMessage` in setUp, to enable more clear output for failed tests.